apps/backend/test/http-ping-bucket/reads.spec.ts

```diff
@@ -0,0 +1,300 @@
+      const monitor = await bootstrap.utils.httpMonitorsUtils.createHttpMonitor({
+        token,
+        projectId: project.id,
+      });
+
+      const now = new Date();
+      const twoDaysAgo = new Date(now.getTime() - 2 * 24 * 60 * 60 * 1000);
+      twoDaysAgo.setMinutes(0, 0, 0);
+      const twoHoursAgo = new Date(now.getTime() - 2 * 60 * 60 * 1000);
+      twoHoursAgo.setMinutes(0, 0, 0);
+      const oneHourAgo = new Date(now.getTime() - 1 * 60 * 60 * 1000);
+      oneHourAgo.setMinutes(0, 0, 0);
+      const twoHoursAgoBucket = {
+        httpMonitorId: monitor.id,
+        timestamp: twoHoursAgo,
+        successCount: 3,
+        failureCount: 4,
+        averageLatencyMs: 200,
+      };
+      const oneHourAgoBucket = {
+        httpMonitorId: monitor.id,
+        timestamp: oneHourAgo,
+        successCount: 5,
+        failureCount: 6,
+        averageLatencyMs: 300,
+      };
+
+      await bootstrap.utils.httpPingBucketUtils.createHttpPingBucket({
+        httpMonitorId: monitor.id,
+        timestamp: twoDaysAgo,
+        successCount: 1,
+        failureCount: 2,
+        averageLatencyMs: 100,
+      });
+      await bootstrap.utils.httpPingBucketUtils.createHttpPingBucket(twoHoursAgoBucket);
+      await bootstrap.utils.httpPingBucketUtils.createHttpPingBucket(oneHourAgoBucket);
+
+      // when
+      const response = await request(bootstrap.app.getHttpServer())
+        .get(`/projects/${project.id}/monitors/${monitor.id}/http_ping_buckets?period=24h`)
+        .set('Authorization', `Bearer ${token}`);
+
+      // then
+      expect(response.status).toBe(200);
+      expect(response.body).toHaveLength(24);
+
+      const expectedOneHourAgoBucket = response.body[1];
+      expect(expectedOneHourAgoBucket).toMatchObject({
+        successCount: oneHourAgoBucket.successCount,
+        failureCount: oneHourAgoBucket.failureCount,
+        averageLatencyMs: oneHourAgoBucket.averageLatencyMs,
+        timestamp: oneHourAgo.toISOString(),
+      });
+
+      const expectedTwoHoursAgoBucket = response.body[2];
+      expect(expectedTwoHoursAgoBucket).toMatchObject({
+        successCount: twoHoursAgoBucket.successCount,
+        failureCount: twoHoursAgoBucket.failureCount,
+        averageLatencyMs: twoHoursAgoBucket.averageLatencyMs,
+        timestamp: twoHoursAgo.toISOString(),
+      });
+
+      const nullBuckets = response.body.filter((bucket) => bucket === null);
+      expect(nullBuckets).toHaveLength(22);
+    });
+
+    it('gets virtual bucket for current hour', async () => {
+      // given
+      const { token, project } = await bootstrap.utils.generalUtils.setupAnonymous();
+      const monitor = await bootstrap.utils.httpMonitorsUtils.createHttpMonitor({
+        token,
+        projectId: project.id,
+      });
+      const currentHour = new Date();
+      currentHour.setMinutes(0, 0, 0);
+      await bootstrap.utils.httpPingUtils.createHttpPing({ httpMonitorId: monitor.id });
```

Comment:
Can we add another ping to confirm that all the aggregations work? That average really averages and that failure count also matches?

Reply:
done